roles/cluster_infra/templates/resources.tf.j2

```diff
@@ -111,12 +127,50 @@ resource "openstack_compute_instance_v2" "control" {
     name = "{{ cluster_network }}"
   }
   security_groups = ["${openstack_networking_secgroup_v2.secgroup_slurm_cluster.name}"]
-  # Use cloud-init to inject the SSH keys
+
+  # root device:
+  block_device {
```

Comment:
I get the `home` and `state` volumes, but I don't understand why we need the root volume defined now. Should we let it be determined by the flavor rather than forcing `destination_type = "local"`?

Reply:
If you don't include a `block_device{}` block for root then you get:
```
│ Error: Error creating OpenStack server: Bad request with: [POST https://arcus.openstack.hpc.cam.ac.uk:8774/v2.1/servers], error message: {"badRequest": {"code": 400, "message": "Block Device Mapping is Invalid: Boot sequence for the instance and image/block device mapping combination is not valid."}}
```
There's an [example in the docs](https://registry.terraform.io/providers/terraform-provider-openstack/openstack/latest/docs/resources/compute_instance_v2#boot-instance-create-volume-and-attach-volume-as-a-block-device) specifically for this case and it does specify the root volume.

Reply:
Also, trying with a block_device{} specified for the root but not specifying the documented-as-optional `destination_type` fails with:
```
Error: unknown block device destination type 
```

Reply:
Yeah, I think this is an API requirement here. I think before we would always have been doing this image/local combo anyways, so there is no change here, AFAICT anyways.